### ISSUE TITLE
compilation failure

### ISSUE THREAD

**Issue by phykos** [digital-fabric/polyphony#43]
Ruby: MRI 2.7.2p137 (manually compiled)
OS: GNU/Linux
C compiler: gcc 10.2.0

```
$ sudo gem i polyphony

Fetching polyphony-0.45.4.gem
Building native extensions. This could take a while...
ERROR:  Error installing polyphony:
	ERROR: Failed to build gem native extension.

    current directory: /usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony
/usr/bin/ruby -I /usr/lib/ruby/2.7.0 -r ./siteconf20201003-3226-b8zqx0.rb extconf.rb
checking for unistd.h... yes
checking for linux/aio_abi.h... yes
checking for sys/select.h... yes
checking for port_event_t in poll.h... no
checking for sys/epoll.h... yes
checking for sys/event.h... no
checking for port_event_t in port.h... no
checking for sys/resource.h... yes
creating Makefile

current directory: /usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony
make "DESTDIR=" clean

current directory: /usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony
make "DESTDIR="
compiling event.c
compiling fiber.c
compiling libev.c
In file included from libev.c:2:
../libev/ev.c:511:48: warning: "/*" within comment [-Wcomment]
  511 | /*#define MIN_INTERVAL  0.00000095367431640625 /* 1/2**20, good till 2200 */
      |                                                 
In file included from libev.c:2:
../libev/ev.c: In function ‘ecb_binary32_to_binary16’:
../libev/ev.c:1302:13: warning: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘int’ [-Wsign-compare]
 1302 |       if (e < (14 - 24)) /* might not be sharp, but is good enough */
      |             ^
In file included from libev.c:2:
../libev/ev.c: At top level:
../libev/ev.c:1867:31: warning: ‘ev_default_loop_ptr’ initialized and declared ‘extern’
 1867 |   EV_API_DECL struct ev_loop *ev_default_loop_ptr = 0; /* needs to be initialised to make it a definition despite extern */
      |                               ^~~~~~~~~~~~~~~~~~~
../libev/ev.c: In function ‘array_nextsize’:
../libev/ev.c:1978:19: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 1978 |   if (elem * ncur > MALLOC_ROUND - sizeof (void *) * 4)
      |                   ^
In file included from ../libev/ev.c:2745,
                 from libev.c:2:
../libev/ev_linuxaio.c: In function ‘linuxaio_poll’:
../libev/ev_linuxaio.c:498:10: warning: suggest explicit braces to avoid ambiguous ‘else’ [-Wdangling-else]
  498 |       if (expect_false (res < 0))
      |          ^
In file included from libev.c:2:
../libev/ev.c: In function ‘ev_io_start’:
../libev/ev.c:4017:34: warning: suggest parentheses around arithmetic in operand of ‘|’ [-Wparentheses]
 4017 |   fd_change (EV_A_ fd, w->events & EV__IOFDSET | EV_ANFD_REIFY);
      |                        ~~~~~~~~~~^~~~~~~~~~~~~
../libev/ev.c: At top level:
../libev/ev.c:5271:27: warning: "/*" within comment [-Wcomment]
 5271 | /* EV_STAT     0x00001000 /* stat data changed */
      |                            
../libev/ev.c:5272:27: warning: "/*" within comment [-Wcomment]
 5272 | /* EV_EMBED    0x00010000 /* embedded event loop needs sweep */
      |                            
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may have been intended to silence earlier diagnostics
compiling libev_backend.c
compiling polyphony.c
compiling polyphony_ext.c
compiling queue.c
compiling ring_buffer.c
compiling thread.c
compiling tracing.c
linking shared-object polyphony_ext.so
/usr/bin/ld: thread.o:/usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony/thread.c:7: multiple definition of `ID_ivar_result'; fiber.o:/usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony/fiber.c:6: first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:262: polyphony_ext.so] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4 for inspection.
Results logged to /usr/lib/ruby/gems/2.7.0/extensions/aarch64-linux/2.7.0/polyphony-0.45.4/gem_make.out
```

Perhaps the big hint may be these lines:
```
/usr/bin/ld: thread.o:/usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony/thread.c:7: multiple definition of `ID_ivar_result'; fiber.o:/usr/lib/ruby/gems/2.7.0/gems/polyphony-0.45.4/ext/polyphony/fiber.c:6: first defined here
```

**Comment by ciconia** [comment]
Thanks for reporting this! I've just released version 0.45.5. Please try installing the gem and let me know if works.

**Comment by phykos** [comment]
Works!

**Comment by andrii-rymar** [comment]
Hey, having the similar issue with 0.46.0 on macOS 10.15.6:

```
compiling liburing.c
compiling playground.c
compiling polyphony.c
compiling polyphony_ext.c
compiling queue.c
compiling ring_buffer.c
compiling runqueue.c
compiling runqueue_ring_buffer.c
compiling thread.c
compiling tracing.c
linking shared-object polyphony_ext.bundle
duplicate symbol '_ID_ivar_result' in:
    fiber.o
    thread.o
ld: 1 duplicate symbol for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [polyphony_ext.bundle] Error 1

make failed, exit code 2
```

**Comment by ciconia** [comment]
Should be fixed in 0.46.1. Please try again and let me know if you have any more issues. Thanks!

**Comment by andrii-rymar** [comment]
Nice, thanks!

**Comment by primeapple** [comment]
Hey there, I'm running into a similar issue with MacOS 11.3.1, Ruby 2.7.2
```
Building native extensions. This could take a while...
ERROR:  Error installing polyphony:
	ERROR: Failed to build gem native extension.

    current directory: /Users/me/.rvm/gems/ruby-2.7.2@test/gems/polyphony-0.53.0/ext/polyphony
/Users/me/.rvm/rubies/ruby-2.7.2/bin/ruby -I /Users/me/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0 -r ./siteconf20210510-5414-1e17gys.rb extconf.rb
checking for linux/aio_abi.h... no
checking for sys/select.h... yes
checking for port_event_t in poll.h... no
checking for sys/epoll.h... no
checking for sys/event.h... yes
checking for sys/queue.h... yes
checking for port_event_t in port.h... no
checking for sys/resource.h... yes
creating Makefile

current directory: /Users/me/.rvm/gems/ruby-2.7.2@test/gems/polyphony-0.53.0/ext/polyphony
make "DESTDIR=" clean

current directory: /Users/me/.rvm/gems/ruby-2.7.2@test/gems/polyphony-0.53.0/ext/polyphony
make "DESTDIR="
compiling backend_io_uring.c
compiling backend_io_uring_context.c
compiling backend_libev.c
backend_libev.c:804:11: error: implicit declaration of function 'splice' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    len = splice(src_fptr->fd, 0, dest_fptr->fd, 0, NUM2INT(maxlen), 0);
          ^
backend_libev.c:862:11: error: implicit declaration of function 'splice' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    len = splice(src_fptr->fd, 0, dest_fptr->fd, 0, NUM2INT(maxlen), 0);
          ^
2 errors generated.
make: *** [backend_libev.o] Error 1

make failed, exit code 2
```

**Comment by ciconia** [comment]
@primeapple Thanks for reporting this, fixed in 1245e4b. I've just released version 0.53.1 which includes the fix. Please try reinstalling and let me know if you have any further problems.
